src/Symfony/Component/JsonStreamer/ValueTransformer/ValueObjectToScalarValueTransformer.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\JsonStreamer\ValueTransformer;
+
+use BcMath\Number;
+use Symfony\Component\TypeInfo\Type;
+
+/**
+ * Transforms a value object to its related scalar value during stream writing.
+ *
+ * Does nothing if the value is not a valid value object.
+ *
+ * @author Mathias Arlaud <mathias.arlaud@gmail.com>
+ *
+ * @experimental
+ */
+final class ValueObjectToScalarValueTransformer implements ValueTransformerInterface
```

Comment:
Why making a single ValueObjectToScalarValueTransformer that only handles 3 king of objects anyway ? This is not the way we do that in other components. Can't we have separate transformers for DateTimeInterface, bcmath and gmp instead ?

Reply:
Indeed, that'll be much easier, I updated in that way. I kept only two transformers, one for date times and one for numbers, as dealing with `Number` and `\GMP` is very similar.